### PR TITLE
Movies

### PR DESCRIPTION
**Pull request by gduck**
scrape for filling movie db

### CODE REVIEW

File: app/assets/javascripts/movies.js.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (h8rry) [inline app/assets/javascripts/movies.js.coffee]: can delete file

File: app/assets/stylesheets/movies.css.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Movies controller here.
```
Comment (h8rry) [inline app/assets/stylesheets/movies.css.scss]: can delete file

File: app/helpers/movies_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module MoviesHelper
```
Comment (h8rry) [inline app/helpers/movies_helper.rb]: can delete file

File: test/helpers/movies_helper_test.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+require 'test_helper'
```
Comment (h8rry) [inline test/helpers/movies_helper_test.rb]: can delete file

File: test/models/movie_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (h8rry) [inline test/models/movie_test.rb]: can delete file

File: test/fixtures/movies.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (h8rry) [inline test/fixtures/movies.yml]: can delete file

Comment (h8rry) [inline test/fixtures/movies.yml]: should have done `rails new <app_name> -BT` in the beginning. Now, it's generating many test files. If you want to test, use rspec.

File: test/controllers/movies_controller_test.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (h8rry) [inline test/controllers/movies_controller_test.rb]: can delete file

File: lib/tasks/scrape.rake [hunk 8]
```diff
@@ -138,9 +138,16 @@ namespace :scrape do
     puts article_links
     number_articles = article_links.count
     puts "number of article links " + number_articles.to_s
+  end
 
+  desc "scrape movie of the year, store it"
+  task :movies => :environment do
```
Comment (h8rry) [inline lib/tasks/scrape.rake]: how come this is empty

File: lib/tasks/scrapeMovies.rake [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+  desc "scrape oscars.org for list of best movies"
+  task :movies => :environment do |t, args|
+
+    require 'open-uri'
+    require 'nokogiri'
+
+    #  can call like this if you want
+    # rake scrapeMovies:movies year="2000"
+    year = ENV['year']
```
Comment (h8rry) [inline lib/tasks/scrapeMovies.rake:14]: are we using this?

File: lib/tasks/scrapeMovies.rake [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    # puts "year is #{year}"
+
+    for year in 1940..2015 do
+
+      url = "http://www.imdb.com/event/ev0000003/" + year.to_s
```
Comment (h8rry) [inline lib/tasks/scrapeMovies.rake]: this can be `url = "http://www.imdb.com/event/ev0000003/#{year}"`

File: lib/tasks/scrapeMovies.rake [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      end
+
+
+      data_movies = "blockquote > blockquote:nth-child(2) > div > strong:nth-child(2) > a:nth-child(1)"
+      movies = html_doc.css(data_movies)
+
+      count = movies.count
+      if count == 0 then
+        puts "big problem - no movies"
+        return
```
Comment (h8rry) [inline lib/tasks/scrapeMovies.rake:38]: `return` or `next`? `return` will break outside of the loop

Comment (gduck) [inline lib/tasks/scrapeMovies.rake:38]: I would prefer this scrape stops if there is a missing page so I think return is ok..?

File: lib/tasks/scrapeMovies.rake (hunk 9 above)
Comment (gduck) [inline lib/tasks/scrapeMovies.rake:14]: I was testing with individual years ie rake scrapeMovies:movies year="2000"
This is handy code I might want to refer to again - I didn't want to delete it :/

**Pull request merged**